onnxruntime/core/providers/dml/DmlExecutionProvider/src/MLOperatorAuthorImpl.cpp

```diff
@@ -1120,57 +1120,64 @@
-    {
-        // Populate the mappings between DML in/outs and kernel in/outs.  By default they are the same.
-        if (dmlProperties && dmlProperties->kernelInputIndices)
-        {
-            m_graphNodeCreateInfo->kernelInputIndices.insert(
-                m_graphNodeCreateInfo->kernelInputIndices.begin(),
-                dmlProperties->kernelInputIndices,
-                dmlProperties->kernelInputIndices + dmlProperties->dmlInputCount);
-        }
-        else
-        {
-            m_graphNodeCreateInfo->kernelInputIndices.resize(dmlProperties ? dmlProperties->dmlInputCount : GetInputCount());
-            std::iota(m_graphNodeCreateInfo->kernelInputIndices.begin(), m_graphNodeCreateInfo->kernelInputIndices.end(), 0);
-        }
-
-        if (dmlProperties && dmlProperties->kernelOutputIndices)
-        {
-            m_graphNodeCreateInfo->kernelOutputIndices.insert(
-                m_graphNodeCreateInfo->kernelOutputIndices.begin(),
-                dmlProperties->kernelOutputIndices,
-                dmlProperties->kernelOutputIndices + dmlProperties->dmlOutputCount);
-        }
-        else
-        {
-            m_graphNodeCreateInfo->kernelOutputIndices.resize(dmlProperties ? dmlProperties->dmlOutputCount : GetOutputCount());
-            std::iota(m_graphNodeCreateInfo->kernelOutputIndices.begin(), m_graphNodeCreateInfo->kernelOutputIndices.end(), 0);
-        }
-
-        m_graphNodeCreateInfo->allowHalfPrecisionComputation = dmlProperties ? dmlProperties->allowHalfPrecisionComputation : true;
-    }
-
     HRESULT STDMETHODCALLTYPE DmlGraphOpKernelInfoWrapper::SetDmlOperator(
-        IDMLOperator* op,
-        _In_ const DML_OPERATOR_DESC* desc,
-        _In_opt_ const MLOperatorKernelDmlProperties* dmlProperties) const noexcept
+        _In_ const MLOperatorGraphDesc* operatorGraphDesc) const noexcept
     {
         ORT_TRY
         {
-          ML_CHECK_BOOL(op != nullptr);
-          ML_CHECK_BOOL(dmlProperties != nullptr);
+            ML_CHECK_BOOL_MSG(operatorGraphDesc != nullptr, "OperatorGraphDesc can't be null while creating operator kernel for DML Graph");
+            // Either nodesAsOpDesc or nodesIDMLOperator can be present.
+            assert(!(operatorGraphDesc->nodesAsOpDesc != nullptr && operatorGraphDesc->nodesAsIDMLOperator != nullptr));
+            assert(operatorGraphDesc->nodesAsOpDesc != nullptr || operatorGraphDesc->nodesAsIDMLOperator != nullptr);
 
-          m_graphNodeCreateInfo->initialized = true;
+            if (operatorGraphDesc->nodesAsOpDesc)
+            {
+                m_graphNodeCreateInfo->nodesAsOperatorDesc = std::make_optional<std::vector<std::unique_ptr<AbstractOperatorDesc>>>();
+                for (uint32_t nodeIndex = 0; nodeIndex < operatorGraphDesc->nodeCount; nodeIndex++) 
+                {
+                    auto node = operatorGraphDesc->nodesAsOpDesc + nodeIndex;
```

Comment:
>auto node = operatorGraphDesc->nodesAsOpDesc + nodeIndex; [](http://example.com/codeflow?start=20&length=57)

`auto* node = auto node = operatorGraphDesc->nodesAsOpDesc[nodeIndex];` would be a little clearer/idiomatic.